### PR TITLE
3 get api categories

### PR DESCRIPTION
**Pull request by elifalinli**
please see the most recent code!

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+
+app.use((err, req, res, next) => {
+  if (err.status === 404) {
+    res.status(404).send({ msg: "404: does not exist!" });
+  } else {
+    console.log(err);
+    res.status(500).send({ msg: "Ooopss! Server error!" });
+  }
+});
```
Comment (NRMSMITH) [inline app.js:18]: much cleaner 👍

File: app.test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            expect(categories).toHaveLength(4);
+            categories.forEach((category) => {
+                expect(category).toMatchObject({
+                        slug: expect.any(String),
+                        description: expect.any(String)
+                    });
+                });
+        });
+    })
+  });
```
Comment (NRMSMITH) [inline app.test.js]: Please test for a bad pathway! :)

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
Nearly there Elif! Please test for a 404 bad pathway too :)

-> Please do not 'close' this pull request, instead do your changes locally, then add, commit and push to this branch and then your pull request will update!! Just make sure you send it again through /nchelp pr tomorrow after the lecture! :)

File: app.test.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          });
+        });
+      });
+  });
+  it("status 404: , responds with an error message when passed a non-existent path", () => {
+    return request(app) 
+      .get("/api/categoris")
+      .expect(404)
+      .then(({body}) => {
+        console.log(body)
```
Comment (itskatherine) [inline app.test.js:36]: Remember to get rid of console.logs for PRs 👀

**Review by itskatherine — APPROVED** [review]
Looks good Elif just get rid of those pesky console logs then good to merge! 👍

**Pull request merged**